build/document-extractor.js

```diff
@@ -71,6 +71,69 @@ function extractSections($) {
     sections.push(...subSections);
     flaws.push(...subFlaws);
   }
+
+  // Check for an mutate possible duplicated IDs.
+  // If a HTML document has...:
+  //
+  //   <h2 id="Examples">Check these examples</h2>
+  //   ...
+  //   <h2 id="examples">Examples</h2>
+  //
+  // then this can cause various problems. For example, the anchor links
+  // won't work. The Table of Contents won't be able to do a loop with unique
+  // `key={section.id}` values.
+  // The reason we need to loop through to get a list of all existing IDs
+  // first is because we might have this:
+  //
+  //  <h2 id="foo">Foo X</h2>
+  //  <h2 id="foo">Foo Y</h2>
+  //  <h2 id="foo_2">Foo Z</h2>
+  //
+  // So when you encounter `<h2 id="foo">Foo Y</h2>` you'll know that you
+  // can't suggest it to be `<h2 id="foo_2">Foo Y</h2>` because that ID
+  // is taken by another one, later.
+  const allIDs = new Map();
```

Comment:
Both `allIDs` here and `seenIDs` below could be simplified to a set, since with the current approach, if an ID is in the `Map` , it'll always have a value of 1 or greater, so truthy, and we never use the value as a number, just for its truthiness.
```suggestion
  const allIDs = new Set();
```